### ISSUE TITLE
[CLI]: The loss curve keeps jumping around when there are lots of entries

### ISSUE THREAD

**Issue by BlinkDL** [wandb/client#3687]
### Describe the bug

Please take a look at https://wandb.ai/blinkdl/RWKV-v2-RNN-Pile

Watch the latest run for 10 minutes and you will see the loss curve keeps jumping around.

Adding a EMA of 0.99 does not help.

### Environment

WandB version: 0.12.16

OS: Ubuntu 20

Python version: 3.9.9

**Comment by ramit-wandb** [comment]
Hi @BlinkDL,

Are you talking about the sharp spikes present in the graph? They seem to be coming from the fact that your gradients seem to be exploding, taking your loss to NaN. Your model seems to be logging a NaN loss, you might want to look into why that is happening.

Thanks,
Ramit

**Comment by BlinkDL** [comment]
> Hi @BlinkDL,
> 
> Are you talking about the sharp spikes present in the graph? They seem to be coming from the fact that your gradients seem to be exploding, taking your loss to NaN. Your model seems to be logging a NaN loss, you might want to look into why that is happening.
> 
> Thanks, Ramit

No. Not the NaNs. Spend 10 minutes watching the latest run, and you will find the loss curve changes a lot after each update: https://wandb.ai/blinkdl/RWKV-v2-RNN-Pile/runs/3fo1pedl?workspace=user-

**Comment by BlinkDL** [comment]
> I see. I just looked over your project, specifically [this run](https://wandb.ai/blinkdl/RWKV-v2-RNN-Pile/runs/7t7gu905) and I see what you are talking about. For performance reasons, we downsample our runs when there are a lot of points being logged and interpolate between the sampled runs. You are seeing these changes in your curves because of this sampling.
> 
> Here is a [link](https://docs.wandb.ai/ref/app/features/panels/line-plot/sampling#bucketing) to our docs that discuss this in more detail.
> 
> Thanks, Ramit

Could you downsample the loss using a box filter instead of simple subsampling?

**Comment by ramit-wandb** [comment]
I'll definitely go ahead and create a feature request for this.

**Comment by orcinus** [comment]
A year later, and still no changes here?

**Comment by anmolmann** [comment]
This issue has been automatically closed because it has been inactive for a while. If you still intend to pursue this, please leave a comment and we can reopen it.
